src/TicketController.java

```diff
@@ -0,0 +1,63 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.Scanner;
+
+public class TicketController {
+    public static void main(String[] args) throws ParseException {
+        TicketMachine ticketMachine = new TicketMachine();
+
+        Passenger[] passengers = new Passenger[3];
+        passengers[0] = new Passenger("Wojciech", "Kostecki");
+        passengers[1] = new Passenger("Rafał", "Bobko");
+        passengers[2] = new Passenger("Damian", "Podbilski");
+
+        ticketMachine.buyTicket(passengers[0], 30);
+        ticketMachine.buyTicket(passengers[1], 20);
+        ticketMachine.buyTicket(passengers[2], 15);
+
+        printPassengerWithoutTicket(validateTickets(passengers));
+    }
+
+    private static Passenger[] validateTickets(Passenger[] passengers) throws ParseException {
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Data i godzina kontroli w formacie: yyyy-MM-dd HH:mm");
+        String date = scanner.nextLine();
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+        Date controlDate = sdf.parse(date);
+
+        int passengersWithoutTicket = 0;
+
+        for (Passenger passenger : passengers) {
+            LocalDateTime ticketDate = passenger.getTicket().getBuyingTime().
+                    plusMinutes(passenger.getTicket().getValidityTime());
+            Date ticketExpiryDate = Date.from(ticketDate.atZone(ZoneId.systemDefault()).toInstant());
+            if (controlDate.after(ticketExpiryDate)) {
+                passengersWithoutTicket++;
+            }
+        }
+
+        Passenger[] withoutTicket = new Passenger[passengersWithoutTicket];
+
+        for (int i = 0, j = 0; i < passengers.length; i++) {
```

Comment:
I am guessing that this code works but it is not optimal. You are iterating through array twice  when you could do all of it in one go. Stop using arrays and switch to lists.

Reply:
With the list it would even be easier, but the recommendation shows that there should be a passenger table and the method returns the table.